src/formatDate.js

```diff
@@ -8,7 +8,70 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+  const dateParts = date.split(oldSeparator);
+  let year = '';
+  let month = '';
+  let day = '';
+  const formatDateParts = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i][0]) {
+      case 'Y':
+        year = dateParts[i];
+        break;
+      case 'M':
+        month = dateParts[i];
+        break;
+      case 'D':
+        day = dateParts[i];
+        break;
+      default:
+        throw new Error('not correct format');
+    }
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    switch (toFormat[i][0]) {
+      case 'Y':
+        formatDateParts.push(getFormatYear(year, toFormat[i]));
+        break;
+      case 'M':
+        formatDateParts.push(month);
+        break;
+      case 'D':
+        formatDateParts.push(day);
+        break;
+      default:
+        throw new Error('not correct format');
+    }
+  }
+
+  return formatDateParts.join(newSeparator);
```

Comment:
The implementation of the `formatDate` function seems to follow the task's requirements. However, it lacks input validation which could lead to unexpected behavior if the function is passed incorrect or incomplete arguments. Consider adding checks to ensure that `date`, `fromFormat`, and `toFormat` are provided and that they are of the expected types and formats before proceeding with the transformation.